tests/interfaces/protocols/handlers/protocols-query.spec.ts

```diff
@@ -93,6 +100,88 @@ describe('ProtocolsQueryHandler.handle()', () => {
       expect(reply2.entries?.length).to.equal(3); // expecting all 3 entries written above match the query
     });
 
+    it('should return protocols with normalized URIs matching the query', async () => {
+      const alice = await TestDataGenerator.generatePersona();
+
+      // setting up a stub method resolver
+      TestStubGenerator.stubDidResolver(didResolver, [alice]);
+
+      // configure several protocols, all matching `example.com`
+      const protocolDefinition = emailProtocolDefinition;
+      const protocol1 = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com',
+        protocolDefinition
+      });
+      const protocolTrailingSlash = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com/',
+        protocolDefinition
+      });
+      const protocolParams = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'example.com/?foo=bar',
+        protocolDefinition
+      });
+      const protocolCapitalized = await TestDataGenerator.generateProtocolsConfigure({
+        requester : alice,
+        protocol  : 'EXAMPLE.Com',
+        protocolDefinition
+      });
+
+      // Sort messages into lexicographic order. We only allow protocol overwrites if the new message CID
+      // has higher lexicographic value.
```

Comment:
wow really?  that seems like an odd way of restricting things.

we dont sort based on the `dateModified` or anything like that?

Reply:
Yea I don't understanding why. @csuwildcat Why do we do this? I know I pinged in Slack, but asking here too so we can get an opened source'd explanation.

Reply:
@dcrousso and @diehuxx, the comment is not entirely correct. We only resort to lexicographical order if all messages for the same ID are written with the exact same timestamp, which is the case here in the context of testing, but not a commonly expected scenario in the real-world.

Reply:
Ahh okay that makes more sense. Was very confused. Thanks!